packages/web/src/components/audio-pill/AudioPill.tsx

```diff
@@ -0,0 +1,82 @@
+import { cloneElement, useMemo } from 'react'
+
+import {
+  BNWei,
+  accountSelectors,
+  formatWei,
+  isNullOrUndefined,
+  useSelectTierInfo,
+  walletSelectors
+} from '@audius/common'
+import BN from 'bn.js'
+import cn from 'classnames'
+import { useSelector } from 'react-redux'
+
+import IconNoTierBadge from 'assets/img/tokenBadgeNoTier.png'
+import Skeleton from 'components/skeleton/Skeleton'
+import { audioTierMapPng } from 'components/user-badges/UserBadges'
+
+import styles from './AudioPill.module.css'
+
+const { getAccountUser } = accountSelectors
+const { getAccountTotalBalance } = walletSelectors
+
+type AudioPillProps = {
+  className?: string
+}
+
+/**
+ * Pulls balances from account and wallet selectors. Will prefer the wallet
+ * balance once it has loaded. Otherwise, will return the account balance if
+ * available. Falls back to 0 if neither wallet or account balance are available.
+ */
+const useTotalBalanceWithFallback = () => {
```

Comment:
could this fn be shared between web and mobile?

Reply:
done